components/scg/src/scg.js

```diff
@@ -690,30 +703,99 @@ SCg.Editor.prototype = {
         });
 
         this.toolDelete().click(async function () {
-            if (self.scene.selected_objects.length > 0) {
-                if (self.scene.selected_objects.length > 1) {
-                    const cantDelete = [];
-                    const deletableObjects = await Promise.all(self.scene.selected_objects.filter(obj => obj.sc_addr).map(async (obj) => {
-                        const canDelete = await self.checkCanDelete(obj.sc_addr);
-                        if (canDelete) {
-                            cantDelete.push(obj);
-                        } else {
-                            return obj;
-                        }
-                    })).then(arr => arr.filter(Boolean));
+            if (!self.scene.selected_objects.length) return;
 
-                    function diffArray(arr1, arr2) {
-                        return arr1.filter(item => !arr2.includes(item));
-                    }
-                    self.scene.deleteObjects(diffArray(self.scene.selected_objects, cantDelete));
-                    self.scene.addDeletedObjects(deletableObjects);
-                } else {
-                    self.scene.deleteObjects(self.scene.selected_objects);
-                    self.scene.addDeletedObjects(self.scene.selected_objects);
+            DeleteButtons.init();
+
+            self.scene.setModal(SCgModalMode.SCgModalType);
+            self.onModalChanged();
+            var tool = $(this);
+
+            function stop_modal() {
+                tool.popover('destroy');
+                self.scene.setEditMode(SCgEditMode.SCgModeSelect);
+                self.scene.setModal(SCgModalMode.SCgModalNone);
+            }
+
+            el = $(this);
+            el.popover({
+                content: self.delete_panel_content,
+                container: container,
+                html: true,
+                delay: {
+                    show: 500,
+                    hide: 100
+                }
+            }).popover('show');
+            cont.find('.popover-content').append(
+                '<button id="scg-close-btn" type="button" class="close scg-close-btn-fragments-window">&times;</button>'
+            );
+
+            if (self.scene.selected_objects.length === 1) {
+                if(!self.scene.selected_objects[0].sc_addr) {
```

Comment:
Please fix **everywhere** in file
```suggestion
                if (!self.scene.selected_objects[0].sc_addr) {
```